scripts/species.py

```diff
@@ -0,0 +1,657 @@
+                    "habitat_yearround_pct",
+                ]
+            ]
+
+            final_df["habitat_yearround_pct"] = final_df["habitat_yearround_pct"].round(
+                4
+            )
+
+            if self.output_format == "parquet":
+                table = pa.Table.from_pandas(final_df)
+                for col in ["item_id", "common_name", "county_id"]:
+                    table = table.set_column(
+                        table.schema.get_field_index(col),
+                        col,
+                        pa.compute.dictionary_encode(table[col]),
+                    )
+                pa.parquet.write_table(table, self.output_dir / "species.parquet")
+            elif self.output_format == "arrow":
+                table = pa.Table.from_pandas(final_df)
+                for col in ["item_id", "common_name", "county_id"]:
+                    table = table.set_column(
+                        table.schema.get_field_index(col),
+                        col,
+                        pa.compute.dictionary_encode(table[col]),
+                    )
+                with (
+                    pa.OSFile(str(self.output_dir / "species.arrow"), "wb") as sink,
+                    pa.RecordBatchFileWriter(sink, table.schema) as writer,
+                ):
+                    writer.write_table(table)
+            else:  # Default to CSV
+                final_df.to_csv(self.output_dir / "species.csv", index=False)
+
+    def run(self) -> None:
+        """Runs the complete habitat data processing workflow."""
+        temp_dir = Path(tempfile.mkdtemp())
+        try:
+            logger.info("Step 1/3: Downloading and extracting habitat data files")
+            tif_files, species_info = self.process_habitat_data(temp_dir)
+
+            species_count = len(species_info)
+            logger.info(
+                "Step 2/3: Beginning habitat analysis for %d species", species_count
+            )
+            results_df = self.analyze_habitat_rasters(tif_files, species_info)
+
+            logger.info("Step 3/3: Saving analysis results")
+            self.save_results(results_df, species_info)
+            logger.info("Analysis complete. Results saved to: %s", self.output_dir)
+
+        finally:
+            if temp_dir.exists():
+                for file in temp_dir.glob("*"):
+                    file.unlink()
+                temp_dir.rmdir()
+
+
+def main() -> None:
+    """Main entry point: loads TOML config, runs the processor."""
+    # --- Configuration Loading (TOML) ---
+    config_path = CONFIG_DIR / "species.toml"  # Correct path
+    try:
+        with config_path.open("rb") as f:
+            config = tomllib.load(f)
+    except FileNotFoundError:
+        logger.error("Configuration file not found: %s", config_path)
+        raise  # Critical error: stop execution
+    except tomllib.TOMLDecodeError as e:
+        logger.error("Error decoding TOML file: %s", e)
+        raise
```

Comment:
https://docs.python.org/3/library/pathlib.html#pathlib.Path.exists

```suggestion
    if config_path.exists():
        config = tomllib.loads(fp.read_text("utf-8"))
    else:
        msg = f"Configuration file not found: {config_path}"
        raise FileNotFoundError(msg)
```

The other error is already doing what you're doing isn't it?